test/11_array.jl

```diff
@@ -638,4 +638,14 @@ end
     @test unique(x) == collect(1500:-1:1)
 end
 
+@testset "categorical() makes a copy of pool and refs" begin
```

Comment:
Put this in arraycommon.jl instead. It would make sense to test with an array supporting `missing` too, and with the `CategoricalArray{T, R}` form with equal and different types, just in case.